StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFlowController.swift

```diff
@@ -49,6 +49,19 @@ extension PaymentSheet {
             }
         }
 
+        var paymentMethodType: PaymentMethodType? {
+            switch self {
+            case let .saved(paymentMethod: paymentMethod, _):
+                return .stripe(paymentMethod.type)
+            case let .new(confirmParams: intentConfirmParams):
+                return intentConfirmParams.paymentMethodType
+            case .applePay, .link:
```

Comment:
Good question. It highlights the ambiguity of what this property is. If it means the type of the PaymentMethod object - e.g. the type of the PaymentMethod that ultimately gets created in the payment process - `.stripe(.link)` is not necessarily accurate, because Link has the "passthrough" mode where the PM type is the underlying pm (e.g. card) and not Link.

In practice that's not what this property is used for. It's for analytics and to determine if the type == sepa, in order to handle special mandate stuff, so returning `.stripe(.link)` is probably more useful for analytics.  I'll add comments to clarify.